src/wallet/api/wallet.cpp

```diff
@@ -1181,6 +1244,31 @@ bool WalletImpl::exportKeyImages(const string &filename, bool all)
   return true;
 }
 
+bool WalletImpl::importKeyImagesFromString(const std::string &data)
+{
+  if (checkBackgroundSync("cannot import key images"))
+    return false;
+  if (!trustedDaemon()) {
+    setStatusError(tr("Key images can only be imported with a trusted daemon"));
+    return false;
+  }
+  try
+  {
+    uint64_t spent = 0, unspent = 0;
+    uint64_t height = m_wallet->import_key_images_string(data, spent, unspent);
+    LOG_PRINT_L2("Signed key images imported to height " << height << ", "
+        << print_money(spent) << " spent, " << print_money(unspent) << " unspent");
+  }
+  catch (const std::exception &e)
+  {
+    LOG_ERROR("Error exporting key images: " << e.what());
+    setStatusError(string(tr("Failed to import key images: ")) + e.what());
+    return false;
+  }
+
+  return true;
+}
```

Comment:
Another general style nitpick: Sometimes you use opening bracket on same line and sometimes on new line, see examples below if it's unclear what I mean. Again both are already present in the API so they're not "wrong", but personally I prefer to not add more chaos and stay consistent within a PR, so I'd use the style from example 2, but again no force.
Example 1 (bracket on same line)
```
something {
}
```
Example 2 (bracket on new line)
```
something
{
}
```

Note: here we have both styles in the same function.